presto-hive/src/main/java/com/facebook/presto/hive/util/TempFileWriter.java

```diff
@@ -0,0 +1,124 @@
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_WRITER_CLOSE_ERROR;
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_WRITER_DATA_ERROR;
+import static com.facebook.presto.orc.OrcEncoding.ORC;
+import static com.facebook.presto.orc.metadata.CompressionKind.LZ4;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.joda.time.DateTimeZone.UTC;
+
+public class TempFileWriter
+        implements Closeable
+{
+    private final PageBuilder pageBuilder;
+    private final OrcWriter orcWriter;
+
+    public TempFileWriter(List<Type> types, OutputStream output)
+    {
+        this.pageBuilder = new PageBuilder(types);
+        this.orcWriter = createOrcFileWriter(output, types);
+    }
+
+    public void writePages(List<Page> pages, int[] pageIndexes, int[] positionIndexes)
+    {
+        checkArgument(pageIndexes.length == positionIndexes.length, "pageIndexes and positionIndexes do not match");
+
+        for (int i = 0; i < pageIndexes.length; i++) {
+            Page page = pages.get(pageIndexes[i]);
+            int position = positionIndexes[i];
+            appendPositionTo(page, position, pageBuilder);
+
+            if (pageBuilder.isFull()) {
+                appendPage(pageBuilder.build());
+                pageBuilder.reset();
+            }
+        }
+
+        if (!pageBuilder.isEmpty()) {
+            appendPage(pageBuilder.build());
+            pageBuilder.reset();
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        try {
+            orcWriter.close();
+        }
+        catch (IOException e) {
+            throw new PrestoException(HIVE_WRITER_CLOSE_ERROR, "Failed to close writer", e);
+        }
+    }
+
+    private void appendPage(Page page)
+    {
+        try {
+            orcWriter.write(page);
+        }
+        catch (IOException e) {
+            throw new PrestoException(HIVE_WRITER_DATA_ERROR, "Failed to write data", e);
+        }
+    }
+
+    public static void appendPositionTo(Page page, int position, PageBuilder pageBuilder)
+    {
+        pageBuilder.declarePosition();
+        for (int i = 0; i < page.getChannelCount(); i++) {
+            Type type = pageBuilder.getType(i);
+            Block block = page.getBlock(i);
+            BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(i);
+            type.appendTo(block, position, blockBuilder);
+        }
+    }
+
+    private static OrcWriter createOrcFileWriter(OutputStream output, List<Type> types)
+    {
+        List<String> columnNames = IntStream.range(0, types.size())
```

Comment:
maybe add index... normally these are `col_N`

Reply:
This is the same way we do it in Raptor -- just the number. It's easy to add a prefix but I don't see the value in doing so.

Reply:
It might be helpful if there is a bug reading this data... not a big deal